### PR TITLE
add "material-nf" icon set, add `icons_format` config option

### PR DESCRIPTION
**Pull request by MaxVerevkin**
With `icons_format`
```toml
icons = "material-nf"
icons_format = " <span size='large' stretch='ultraexpanded' font_family='NotoSans Nerd Font'>{icon}</span> "
```
image

Without `icons_format`
```toml
icons = "material-nf"
```
image
Font used
```
    font pango:JetBrainsMono Nerd Font 10.6
```
### Related issues
- #981
- #1086
### TODO
~~Do not apply `icons_format` for overridden icons? (Not possible with the current implementation of icons deserialization.)~~
- [x] Rename `icons_style` to `icons_format`
- [x] Write documentation
- [x] Make use of more moon icons (for backlight block).
- [ ] Deprecate old `material` icon set?

### CODE REVIEW

File: src/icons.rs [hunk 1]
```diff
@@ -5,280 +5,356 @@ use lazy_static::lazy_static;
...
+        "bat_quarter" => "\u{f57a}",
+        "bat_three_quarters" => "\u{f57f}",
+        "bell" => "\u{f599}",
+        "bell-slash" => "\u{f59a}",
+        "bluetooth" => "\u{f5ae}",
+        "calendar" => "\u{f5ec}",
+        "cogs" => "\u{f992}",
+        "cpu" => "\u{f9c4}",
+        "disk_drive" => "\u{f7c9}",
+        "docker" => "\u{f308}", // nf-linux-docker
```
Comment (ammgws) [inline src/icons.rs:306]: Can you add these comments for the other icons as well? It makes it much easier to figure out the intention.

Can get them from here: https://www.nerdfonts.com/cheat-sheet

File: src/icons.rs [hunk 2]
```diff
@@ -5,280 +5,356 @@ use lazy_static::lazy_static;
...
+        "update" => "\u{e8d7}",
+        "uptime" => "\u{e192}", // Same as time symbol.
+        "volume_empty" => "\u{e04e}",
+        "volume_full" => "\u{e050}",
+        "volume_half" => "\u{e04d}",
+        "volume_muted" => "\u{e04e} \u{e04f}",
+        "xrandr" => "\u{e31e}"
+    };
+
+    // Material an in NerdFont
```
Comment (ammgws) [inline src/icons.rs:284]: "an in"?

Comment (ammgws) [inline src/icons.rs:284]: Can you add a link to https://www.nerdfonts.com/cheat-sheet here?

**Review by ammgws — CHANGES_REQUESTED** [review]

File: src/icons.rs (hunk 2 above)
Comment (MaxVerevkin) [reply src/icons.rs:284]: Sure

Comment (MaxVerevkin) [reply src/icons.rs:284]: I can't even remember what I was trying to write :laughing:

File: src/icons.rs (hunk 1 above)
Comment (MaxVerevkin) [reply src/icons.rs:306]: Will take some time.

**Comment by ammgws** [comment]
Are the Nerd Fonts mappings different than for the official Google material font? I imagine they would be the same if Nerd Fonts is just an aggregator?

**Comment by MaxVerevkin** [comment]
I checked google's [codes](https://github.com/google/material-design-icons/blob/master/font/MaterialIcons-Regular.codepoints) and no, they are not the same. That's weird.

**Comment by ammgws** [comment]
Should icons_format be able to be overridden on a per-block basis like with themes?

**Comment by MaxVerevkin** [comment]
> Should icons_format be able to be overridden on a per-block basis like with themes?

I'm not sure. I think that disabling `icons_format` for icons that were overridden is enough, but needs some more work. Actually, allowing each block to override `icons_format` it simpler. But current implementation of icons' deserialization is messy and should be refactored anyway. I mean, it uses one specific custom macro that is used just once.

**Comment by ammgws** [comment]
> Actually, allowing each block to override icons_format it simpler.

Yeah it might be the way to go. Now too sure but might be useful if for example you wanted to specify some different icon font for the weather block? If it's trivial to implement then I don't really see any downside to allowing users to override it.

> . But current implementation of icons' deserialization is messy and should be refactored anyway. I mean, it uses one specific custom macro that is used just once.

Sure

**Comment by MaxVerevkin** [comment]
> Yeah it might be the way to go.

Done.

**Comment by MaxVerevkin** [comment]
Github actions seem to be down.

File: src/icons.rs [hunk 3]
```diff
@@ -287,9 +317,19 @@ lazy_static! {
         "" => "",
         "backlight_empty" => "\u{e38d}", // nf-weather-moon_new
         "backlight_full" => "\u{e39b}", // nf-weather-moon_full
-        "backlight_partial1" => "\u{e3d1}", // nf-weather-moon_alt_waxing_gibbous_3
-        "backlight_partial2" => "\u{e3ce}", // nf-weather-moon_alt_first_quarter
-        "backlight_partial3" => "\u{e3cb}", // nf-weather-moon_alt_waxing_crescent_4
+        "backlight_1" => "\u{e3d4}", // nf-weather-moon_alt_waxing_gibbous_6
```
Comment (ammgws) [inline src/icons.rs:320]: Update this list too: https://github.com/greshake/i3status-rust/blob/master/themes.md#available-icon-overrides

**Comment by ammgws** [comment]
Looks OK to me. As a last step I want to check the differences between the patched material font, the official Google one and the Nerd Fonts one.

Icon | Patched | Google | NerdFonts
--- | --- | --- | --- | 
bat_charging | `\u{e1a3}` | `??` | `\u{f57d}`
etc

**Comment by MaxVerevkin** [comment]
Actually, NerdFonts' README claims that they are using this material icons, not Google's - https://github.com/Templarian/MaterialDesign

Google's font has only one charging icon - `battery_charging_full e1a3`

I haven't fond any icons that would "work" in any two of those fonts yet.

**Review by ammgws — APPROVED** [review]

**Comment by ammgws** [comment]
Google Material Design icons
https://github.com/google/material-design-icons
https://aur.archlinux.org/packages/ttf-material-design-icons/

Material Design icons by MaterialDesignIcons.com
https://github.com/Templarian/MaterialDesign
https://aur.archlinux.org/packages/ttf-material-design-icons-extended/

Nerd Fonts which includes MaterialDesignIcons.com, possibly patched but codepoints the same?
https://github.com/ryanoasis/nerd-fonts/blob/master/readme.md
https://aur.archlinux.org/packages/nerd-fonts-complete/

"Patched" icons from that gist
https://gist.github.com/draoncc/3c20d8d4262892ccd2e227eefeafa8ef/raw/3e6e12c213fba1ec28aaa26430c3606874754c30/MaterialIcons-Regular-for-inline.ttf

**Comment by MaxVerevkin** [comment]
>  possibly patched but codepoints the same?

From what I can tell, no, codepoints are different.

**Comment by ammgws** [comment]
Looks like it. Let's deal with it separately.

Thanks for this!

**Pull request merged**